notebooks/classify/signature.py

```diff
@@ -0,0 +1,249 @@
+                signatureDict[signature]['featureDict'] = self.countTypes(cleanTable.iloc[index], columns, signatureDict[signature]['presentFeatures'], signatureDict[signature]['featureDict'])
+
+        return signatureDict
+
+    def generateSignatures(self, exactMatches: List[str] = [], regexes: List[str] = [], removeDuplicates:  bool = True, isInvariant: bool = True, removeNaN: bool = True):
+
+        """Function for generating a dictionary of data signature
+
+        Args:
+            exactMatches (List[str], optional): [List of features to remove from the table]. Defaults to [].
+            regexes (List[str], optional): [List of regular expressions. If a feature matches a regular expression, it will be removed from the table.]. Defaults to [].
+            removeDuplicates (bool, optional): [Boolean for whether duplicates should be removed.]. Defaults to True.
+            isInvariant (bool, optional): [Boolean to check whether invariant columns should be removed.]. Defaults to True.
+            removeNaN (bool, optional): [Determines if NaN values should be converted to an empty string]. Defaults to True.
+        """
+
+        self.cleanedData = self.cleanData(exactMatches=exactMatches, regexes=regexes, removeDuplicates=removeDuplicates, isInvariant=isInvariant)
+        self.features = list(self.cleanedData)
+        self.binData = self.binarizeData(self.cleanedData, removeNaN=removeNaN)
+        self.signatureDict = self.generate(self.binData, self.cleanedData)
+        featureMap = {}
+        for key, value in self.signatureDict.items():
+            featureMap[tuple(sorted(list(value['presentFeatures'])))] = key
+        self.featureMap = featureMap
+
+    def findUniques(self, threshold: int = 1):
+        """[Function for finding unique values for a signature. featureValueCounts stores the number of times a value shows up in other signatures. i.e. if the item is ('status', 3): 5, the vlaue status code of 3 shows up in 5 other signatures]
+
+        Args:
+            threshold (int, optional): [Value that defines the threshold for what a unique value within a column should be]. Defaults to 1.
+        """
+        if threshold < 1:
+            print("You set a threshold less than 1, automatically set to 1.")
+            threshold = 1
+
+        uniques = {}
+
+        featureValueCounts = {}
+        for signature, properties in self.signatureDict.items():
+            featureDict = properties['featureDict']
+            uniqueFeatures = []
+            for feature, values in featureDict.items():
+                if 0 < len(values.keys()) <= threshold:
+                    uniqueFeatures.append({feature: list(values.keys())[0]})
+                    key = (feature, list(values.keys())[0])
+                    if key in featureValueCounts:
+                        featureValueCounts[key] += 1
+                    else:
+                        featureValueCounts[key] = 1
+            uniques[signature] = uniqueFeatures
```

Comment:
Not a comment on this function.
Might be nice to include a class method that would iterate through a set of tables or all tables and return a list of DataSignature classes for each table.